### PR TITLE
perl: Add recipe to build perl

### PR DESCRIPTION
**Pull request by jakirkham**
Builds perl on Mac, Linux, and Windows. Taken from conda recipes. ( https://github.com/conda/conda-recipes/tree/master/perl )

### CODE REVIEW

**Comment by jakirkham** [comment]
@asmeurer @dan-blanchard, would you have any interest in being added as maintainers of this recipe?

**Comment by pelson** [comment]
As a heads up, @msarahan added a perl recipe in https://github.com/conda-forge/staged-recipes/pull/101. It would be good to reconcile the two PRs.

**Comment by jakirkham** [comment]
Thanks for the notice. Will combine his changes in the recipe here. Then we can figure out where to go from there.

**Comment by jakirkham** [comment]
So I have incorporated @msarahan's changes for Windows here. Also, went ahead and simplified the versioning so it shouldn't be as difficult to maintain.

**Comment by jakirkham** [comment]
Have added a PR ( https://github.com/msarahan/staged-recipes/pull/1 ) against @msarahan's PR ( https://github.com/conda-forge/staged-recipes/pull/101 ) with merge conflicts resolved. Can sort out the rest from there. Will close once that is resolved.

**Comment by pelson** [comment]
> Will close once that is resolved unless we decide it is simpler to break Perl out and get it in first.

I'm keen to merge once things look good, so for me, a single PR for perl sounds like the preferable option. :+1: on this PR once the CI says :+1: .

**Comment by jakirkham** [comment]
Ok, that sounds good. Building perl takes awhile anyways.

Any idea why AppVeyor says it is canceled? Are you restarting it?

**Comment by pelson** [comment]
> Any idea why AppVeyor says it is canceled? Are you restarting it?

Sorry about that. I cancelled an earlier commit, but apparently that means that all subsequent commits are automatically cancelled. I've restarted.

**Comment by pelson** [comment]
P.S. There are a number of PRs which have put a pretty hefty queue on AppVeyor. It's not ideal, but it should come through in a couple of hours. I'm :+1: on this being merged once that happens.

**Comment by jakirkham** [comment]
Oh, ok. Probably for the best as AppVeyor tends to get pretty backlogged.

**Comment by jakirkham** [comment]
Sounds good.

BTW, how many AppVeyor build can conda-forge do at a time? Is it just 1 or something larger?

**Comment by pelson** [comment]
> BTW, how many AppVeyor build can conda-forge do at a time? Is it just 1 or something larger?

```staged-recipes``` is just 1 I believe. It is my understanding that the feedstocks could all run in parallel theoretically.

**Comment by msarahan** [comment]
+1 for splitting it up.  +1 also for building Perl from source.  When I wrote the bit that just repackages strawberry perl, we didn't have great tools for having bash available explicitly (many people installed cygwin or msys, but that was outside of the recipe, and outside of conda.)

I'd be in favor of keeping the build-from-source recipe from conda-recipes if we can - and add other windows dependencies that make the unix-style tools work.  @mingwandroid is working on getting MSYS2 working in Conda.  That might be best, but there might also be a quicker intermediary.

**Comment by jakirkham** [comment]
Ok, good, glad you had a chance to look at this, @msarahan .

As the consensus, seems to be to split out Perl, I'll just keep working on this here.

Yeah, I think this was initially expecting msys, but am not sure though. Using Strawberry Perl looks like a good solution and I am happy to make use of it for now. When MSYS2 comes around, we can include a new Windows build recipe, up the Perl version number or build number, and release. If all goes well, then no one will really notice the transition.

FWIW we can always add a PR to the Perl feedstock when it gets generated with the changes for a proper Windows build so they don't get lost.

**Comment by jakirkham** [comment]
Passes all tests. :)

**Comment by jakirkham** [comment]
@msarahan, are you ok with this as it is now? Certainly will revisit making a better Windows solution once a clear path exists. For now, it looks like your original proposal does quite nicely.

File: recipes/perl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+source:
+  fn: perl-{{ base_version }}.tar.gz  #[unix]
+  fn:  strawberry-perl-{{ version }}-32bit-portable.zip  #[win]
+  url: http://www.cpan.org/src/5.0/perl-{{ base_version }}.tar.gz  #[unix]
+  url:  http://strawberryperl.com/download/5.20.3.1/strawberry-perl-{{ version }}-32bit-portable.zip  #[win]
+  md5: d647d0ea5a7a8194c34759ab9f2610cd  #[unix]
+  md5: 7b62cdfc8087b2a0b6a613a4962eebb4  #[win]
+
+build:
+  number: 1
```
Comment (ocefpaf) [inline recipes/perl/meta.yaml]: `0` :stuck_out_tongue_winking_eye:

**Comment by pelson** [comment]
Excellent! Thanks @jakirkham

File: recipes/perl/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/perl/meta.yaml]: Oops. Sorry, I guess it is in now. :/

File: recipes/perl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{% set version = base_version + ".1" %}
+
+
+package:
+  name: perl
+  version: {{ version }}
+
+source:
+  fn: perl-{{ base_version }}.tar.gz  #[unix]
+  fn:  strawberry-perl-{{ version }}-32bit-portable.zip  #[win]
```
Comment (ocefpaf) [inline recipes/perl/meta.yaml]: Just curious. What is this?

File: recipes/perl/meta.yaml (hunk 1 above)
Comment (pelson) [inline recipes/perl/meta.yaml]: Ah. Sorry @ocefpaf. No harm done having a non zero build number though :smile:

**Comment by jakirkham** [comment]
Thanks @pelson and @msarahan.

File: recipes/perl/meta.yaml (hunk 1 above)
Comment (ocefpaf) [inline recipes/perl/meta.yaml]: No problem.  I was just messing with @jakirkham :wink:

File: recipes/perl/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/perl/meta.yaml]: It was a version of Perl that @msarahan found that is built for Windows. Will hold us over until we can get a proper build system for Windows within conda.

Comment (ocefpaf) [inline recipes/perl/meta.yaml]: Cool.

File: recipes/perl/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/perl/meta.yaml]: Will fix this when we next bump the version number.

**Pull request merged**